docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
 Architecture Overview
 =====================
 
+Runhouse is a Python library that allows any application to flexibly and powerfully utilize remote compute
+infrastructure by deploying and calling remote services on the fly. It is principally designed for Machine
+Learning-style workloads (online, offline, training, and inference), where the need for heterogeneous
+remote compute is frequent and flexibility is paramount to minimize costs.
```

Comment:
Love this intro paragraph. When this is all finalized I think we may want to move some of this directly to the docs index page, as an [introduction](https://python.langchain.com/v0.2/docs/introduction/).

I don't think it (the first sections through "Why?") should be hidden in a secondary page in the "Other Topics" nav section